lib/galaxy/app.py

```diff
@@ -576,6 +582,25 @@ def __init__(self, configure_logging=True, use_converters=True, use_display_appl
 
         self._configure_tool_shed_registry()
         self._register_singleton(tool_shed_registry.Registry, self.tool_shed_registry)
+        self._register_celery_galaxy_task_components()
+
+    def _register_celery_galaxy_task_components(self):
+        """
+        Register subtype class instance to support implementation of a user rate limit for execution of celery tasks.
+        The default supertype class does not enforce a user rate limit. This is the case if the celery_user_rate_limit
+        config param is default value of 0.0
```

Comment:
Maybe not specify `0.0` as the default value in this comment? It's already specified in the config schema, and if we choose to modify it we'd have to remember to modify it here as well (we will forget, of course).

Reply:
Do you want me to undo the manual changes to *.sample and *.rst at this point and redo them using make config-rebuild? Also as far as moving migration to a separate commit do you mean to simply edit the migration script and then push it by itself to the same branch under the same PR?

Reply:
Yes, please! Both can be separate commits in the same PR.